ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/wizard/v1/wizard/clientlibs/site/js/wizardview.js

```diff
@@ -297,6 +305,60 @@
             if (tabs && activeIndex > 0) {
                 this.#navigateAndFocusTab(activeIndex - 1);
             }
+            this.hideUnhideNavButtons(this.#_active, this.#getCachedTabs().length);
+        }
+
+        /**
+         * Navigates to the tab at the provided index
+         *
+         * @private
+         * @param {Number} index The index of the tab to navigate to
+         */
+        hideUnhideNavButtons(activeTabIndex, tabsLength) {
+            if(tabsLength === 0 || this.maxEnabledTab <= this.minEnabledTab) {
+                this.getPreviousButtonDiv().classList.add(Wizard.selectors.previousButtonHidden);
+                this.getNextButtonDiv().classList.remove(Wizard.selectors.nextButtonHidden);
+            }
+
+            if(activeTabIndex <= this.minEnabledTab) {
+                this.getPreviousButtonDiv().classList.add(Wizard.selectors.previousButtonHidden);
+            }
+            if(activeTabIndex === this.maxEnabledTab) {
+                this.getNextButtonDiv().classList.add(Wizard.selectors.nextButtonHidden);
+            }
+
+            if(tabsLength > 1 && activeTabIndex > this.minEnabledTab) {
+                this.getPreviousButtonDiv().classList.remove(Wizard.selectors.previousButtonHidden);
+            }
+
+            if(activeTabIndex < this.maxEnabledTab) {
+                this.getNextButtonDiv().classList.remove(Wizard.selectors.nextButtonHidden);
+            }
+        }
+
+        setNavigationRange() {
```

Comment:
this is calculating only at the beginning , this method is called from constructor. It won't work for repeatability or from rules